pkg/sql/feature_derivation.go

```diff
@@ -230,20 +230,7 @@ func InferFeatureColumns(trainIR *ir.TrainClause) error {
 	fcMap := makeFeatureColumnMap(trainIR.Features)
 	fmMap := makeFieldMetaMap(trainIR.Features)
 
-	q := trainIR.Select
-	re, err := regexp.Compile("LIMIT [0-9]+")
-	if err != nil {
-		return err
-	}
-	limitClauseIndexes := re.FindStringIndex(q)
-	if limitClauseIndexes == nil {
-		q = fmt.Sprintf("%s LIMIT %d", q, featureDerivationRows)
-	} else {
-		// TODO(typhoonzero): there may be complex SQL statements that contain multiple
-		// LIMIT clause, using regex replace will replace them all.
-		re.ReplaceAllString(q, fmt.Sprintf("LIMIT %d", featureDerivationRows))
-	}
-
+	q := fmt.Sprintf("SELECT * FROM (\n%s\n) a LIMIT %d", trainIR.Select, featureDerivationRows)
```

Comment:
Seems using subqueries here will cause error: https://travis-ci.com/sql-machine-learning/sqlflow/jobs/260837168, I'll change it back to the regex way for a fast fix. Remove the `LIMIT 10` from train SQL, it works well.

Running the subquery SQL using hive command line also works. Need to find out a solution.